rfcs/0001-messages.md

```diff
@@ -0,0 +1,268 @@
+This proposal is only considering that specific features, additions and upgrades will need to be made when DKG is implemented.
+
+Assuming all participants have a FROST library available we need to define message structures in a way that data can be exchanged between participants. The proposal is a collection of data types so each side can do all the actions needed for a real life situation.
+
+## Definitions
+
+- `dealer`
+- `aggergator`
+- `signer`
+- `nonce`
+- `commitment`
+- 
+
+## Guide-level explanation
+
+We propose a message separated in 2 parts, a header and a payload:
+
+```rust
+struct Message {
+    header: Header,
+    payload: Payload,
+}
+```
+
+`Header` will look as follows:
+
+```rust
+struct Header {
+    msg_type: MsgType,
+    version: MsgVersion,
+    sender: Participant,
+    receiver: Participant, 
+}
+```
+
+While `Payload` will be defined as:
+
+```rust
+enum Payload {
+    DealerBroadcast(MsgDealerBroadcast),
+    Commitments(MsgCommitments),
+    SigningPackage(MsgSigningPackage),
+    SignatureShare(MsgSignatureShare),
+    FinalSignature(MsgFinalSignature),
+}
+```
+
+All the messages and new types will be defined in a new file `src/frost/messages.rs`
+
+## Reference-level explanation
+
+Here we explore in detail the header types and all the message payloads. 
+
+### Header
+
+Fields of the header define new types. Proposed implementation for them is as follows:
+
+```rust
+#[repr(u8)]
+#[non_exhaustive]
+enum MsgType {
+    DealerBroadcast,
+    Commitments,
+    SigningPackage,
+    SignatureShare,
+    FinalSignature,
+}
+
+struct MsgVersion(u8);
+
+struct Participant(u8);
+```
+
+### Payloads
+
+Each payload defines a new message:
+
+```rust
+// Dealer must send this message with initial data to each participant involved.
+// With this, the participant should be able to build a `SharePackage` and use
+//  the `sign()` function.
+// `public_key` can be calculated from the `secret_key`.
+// from `secret_key`.
+struct MsgDealerBroadcast {
+    // The secret key as a frost::Scalar.
+    secret_key: frost::Scalar,
+    // Commitments for the signer as jubjub::AffinePoint.
+    commitment: jubjub::AffinePoint,
+    // The point and verification bytes needed to generate the group public key
+    group_public: (jubjub::AffinePoint, [u8; 32]),
+}
+
+// Each signer participant send to the aggregator the 2 points
+//  needed for commitment building.
+struct MsgCommitments {
+    // The hiding Point.
+    hiding: jubjub::AffinePoint,
+    // The binding Point.
+    binding: jubjub::AffinePoint,
+}
+
+// The aggergator decide what message is going to be signed and
+// send it to each participant with all the commitments collected.
+struct MsgSigningPackage {
+    // The message to be signed as bytes
+    message: &'static [u8],
+    // The collected unpacked commitments for each signer
+    commitments: Vec<(u8, jubjub::AffinePoint, jubjub::AffinePoint),
```

Comment:
I think we might need to specify the size of the vector (or even use a raw structure instead), similarly to the size of the message to be signed, so that the receiver knows how many bytes they should expect.

Reply:
(The ">" symbol is missing at the end.)